File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        left = [0] * len(nums)
        right = [0] * len(nums)

        for i in range(len(nums)):
            if i == 0:
                left[i] = 1
            else:
                left[i] = nums[i - 1] * left[i - 1]

        for i in range(len(nums) - 1, -1, -1):
            if i == len(nums) - 1:
                right[i] = 1
            else:
                right[i] = right[i + 1] * nums[i + 1]

        output = [0] * len(nums)
        for i in range(len(nums)):
            output[i] = right[i] * left[i]
        return output

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public List<List<int>> LevelOrder(TreeNode root) {
        var queue = new Queue<TreeNode>();
        var output = new List<List<int>>();
        if (root != null)
            queue.Enqueue(root);

        while (queue.Count > 0) {
            var currentLevelLength = queue.Count;
            var levelOutput = new List<int>();
            for (int i = 0; i < currentLevelLength; i++) {
                var node = queue.Dequeue();
                levelOutput.Add(node.val);

                if (node.left != null)
                    queue.Enqueue(node.left);

                if (node.right != null)
                    queue.Enqueue(node.right);
            }
            output.Add(levelOutput);
        }

        return output;
    }
}
